### PR TITLE
[Feature] login signup error handling flow

### PR DESCRIPTION
**Pull request by badukwei**
## Summary

Adding error handling methods and error message templates

## Linked Issue

close https://github.com/social-tw/social-tw-website/issues/105

## Details

1. 新增 errorMessage 相關處理流程，如果用戶登入或註冊失敗則會跳出相應的錯誤訊息
2. refactor routes protection 
3. 簡單修改 styles

## Impacted Areas

packages/frontend/src/components/login/AuthForm.tsx
packages/frontend/src/components/modal/ErrorModal.tsx
packages/frontend/src/components/modal/Modal.tsx
packages/frontend/src/components/modal/SignupLoadingModal.tsx
packages/frontend/src/constants/error-messages/loginErrorMessage.ts
packages/frontend/src/contexts/ProtectedRoute.tsx
packages/frontend/src/contexts/User.tsx
packages/frontend/src/hooks/useAutoNavigation.ts
packages/frontend/src/hooks/useErrorMessage.ts
packages/frontend/src/hooks/useLocalStorage.ts
packages/frontend/src/hooks/useLoginWithServer.ts
packages/frontend/src/hooks/useLoginWithWallet.ts
packages/frontend/src/hooks/useSignupWithServer.ts
packages/frontend/src/hooks/useSignupWithWallet.ts
packages/frontend/src/pages/PostCreate.tsx
packages/frontend/src/layouts/AppLayout.tsx
packages/frontend/src/pages/Login.tsx
packages/frontend/src/index.tsx

## Tests

- Error handling

1. 使用錢包進行註冊
2. 在錢包跳出是否同意實選擇拒絕
3. 跳出錯誤訊息

- Route protects

1. 在尚未登入時進入 home or /profile etc. 則會 redirect to /login

### CODE REVIEW

**Comment by badukwei** [comment]
This pr is ready for review. @kittybest @vivianjeng

**Review by kittybest — APPROVED** [review]

File: packages/frontend/src/hooks/useInitUser.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ const useInitUser = (
     logout: () => void
 ) => {
     useEffect(() => {
+        console.log('User loaded')
```
Comment (vivianjeng) [inline packages/frontend/src/hooks/useInitUser.ts:10]: this message can be removed

File: packages/frontend/src/hooks/useLoginWithServer.ts [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+    navigate: (path: string) => void,
+    setErrorCode: (errorCode: keyof typeof ERROR_MESSAGES) => void,
+    setIsLogin: (param: any) => void,
     createUserState: () => Promise<UserState | undefined>
 ) => {
     const loginWithServer = async () => {
         try {
             if (!hashUserId) {
-                throw new Error('No hash user id')
+                throw new Error('MISSING_ELEMENT')
```
Comment (vivianjeng) [inline packages/frontend/src/hooks/useLoginWithServer.ts:16]: I thought the error should be chosen among `ERROR_MESSAGES`
e.g. `ERROR_MESSAGES.MISSING_ELEMENT.code`
or in the future, these strings will easily cause typo
and other developers cannot find the error codes easily, they need to refer to other file or document

**Comment by vivianjeng** [comment]
`main` branch is updated, please resolve the conflicts 🙏🏻

File: packages/frontend/src/hooks/useInitUser.ts (hunk 1 above)
Comment (badukwei) [reply packages/frontend/src/hooks/useInitUser.ts]: Removed

File: packages/frontend/src/hooks/useLoginWithServer.ts (hunk 2 above)
Comment (badukwei) [reply packages/frontend/src/hooks/useLoginWithServer.ts]: Just changed here 
https://github.com/social-tw/social-tw-website/pull/115/commits/2a67e7cb6736d0f22ec7b96c4200e4e269ae1336

File: packages/frontend/src/hooks/useErrorMessage.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from 'react'
+import ERROR_MESSAGES from '../constants/error-messages/loginErrorMessage'
+
+const useErrorMessage = (errorCode: keyof typeof ERROR_MESSAGES | '') => {
+    const errorMessage = useMemo(() => {
+        console.log(errorCode)
```
Comment (vivianjeng) [inline packages/frontend/src/hooks/useErrorMessage.ts:6]: is this message required?

**Comment by badukwei** [comment]
I've resolved all issues in the pr. Please check it @vivianjeng. Thanks!

**Review by vivianjeng — APPROVED** [review]
LGTM

File: packages/frontend/src/hooks/useErrorMessage.ts (hunk 3 above)
Comment (badukwei) [reply packages/frontend/src/hooks/useErrorMessage.ts]: I miss it. sorry

Comment (badukwei) [reply packages/frontend/src/hooks/useErrorMessage.ts]: Just updated it

**Comment by badukwei** [comment]
How should I solve this? Although It has been aproved, merging is still blocked. @vivianjeng 
image

**Review by vivianjeng — APPROVED** [review]

**Pull request merged**